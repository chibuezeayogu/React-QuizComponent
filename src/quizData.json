{
  "quizQuestions": [
    {
      "id": 1,
      "instructionText": "How many continents are there on Planet Earth?",
      "answerOptions": ["5", "6", "7", "8"],
      "answer": "7"
    },
    {
      "id": 2,
      "instructionText": "Greek civilization was flourished in the year?",
      "answerOptions": ["1000 BC", "900 BC", "700 BC", "800 BC"],
      "answer": "800 BC"
    },
    {
      "id": 3,
      "instructionText": "What is the largest continent on earth?",
      "answerOptions": ["Asia", "Africa", "North America", "South America"],
      "answer": "Asia"
    },
    {
      "id": 4,
      "instructionText": "Rome was founded in the year?",
      "answerOptions": ["754 BC", "755 BC", "753 BC", "756 BC"],
      "answer": "753 BC"
    },
    {
      "id": 5,
      "instructionText": " Who was first to reach the South Pole?",
      "answerOptions": ["Capt. Austin", "Capt. White", "Capt. Peter", "Capt. Amundsen"],
      "answer": "Capt. Amundsen"
    },
    {
      "id": 6,
      "instructionText": "Buckingham Palace became the official royal residence during the reign of which British monarch?",
      "answerOptions": ["Queen Elizbeth", "Queen Veronica", "Queen Victoria", "Queen Elizbeth II"],
      "answer": "Queen Victoria"
    },
    {
      "id": 7,
      "instructionText": "In which year Bangladesh was formed as Separate Country?",
      "answerOptions": ["1973", "1971", "1197", "1974"],
      "answer": "1971"
    },
    {
      "id": 8,
      "instructionText": "The Great Pyramid of Giza was built for which Egyptian ruler?",
      "answerOptions": ["Narmer", "Menes", "Cheops", "Khufu"],
      "answer": "Cheops"
    },
    {
      "id": 9,
      "instructionText": "World War I commenced in?",
      "answerOptions": ["1919", "1914", "1913", "1917"],
      "answer": "1914"
    },
    {
      "id": 10,
      "instructionText": "Who is considered the \"Father of the Constitution\"?",
      "answerOptions": ["James Madison", "Peter Madison", "Paul Madison", "Solomon Madison"],
      "answer": "James Madison"
    }
  ]
}