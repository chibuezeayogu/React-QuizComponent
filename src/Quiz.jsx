import React, { Component } from 'react'
import QuizQuestion from './QuizQuestion'
import QuizEnd from './QuizEnd'
import quizData from './quiz_data.json';
/**
 * 
 * @class Quiz
 * @extends Component
 */
class Quiz extends Component {
  constructor(props) {
    super(props)
    this.state = { quiz_position: 1 }
  }

  /**
   * 
   * @method showNextQuestion
   * 
   * @memberof Quiz
   * 
   * @returns { void }
   */
  showNextQuestion = () => {
    this.setState((state) => ({
      quiz_position: state.quiz_position + 1
    }))
  }

  /**
   * 
   * @method handleResetClick
   * 
   * @memberof Quiz
   * 
   * @returns { void }
   */
  handleResetClick = () => {
    this.setState({ quiz_position: 1 })
  }

  render() {
    const {
      state: { quiz_position },
      handleResetClick,
      showNextQuestion
    } = this;
    const isQuizEnd = (quiz_position - 1) === quizData.quiz_questions.length;
    return (
      <div>
        {isQuizEnd ?
          <QuizEnd resetClickHandler={handleResetClick} /> :
          <QuizQuestion
            quiz_question={quizData.quiz_questions[quiz_position - 1]}
            showNextQuestionHandler={showNextQuestion}
          />
        }
      </div>
    )
  }
}

export default Quiz;
