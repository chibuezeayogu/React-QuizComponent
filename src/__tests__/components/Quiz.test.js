import React from 'react';
import { shallow } from 'enzyme';

import Quiz from '../../Quiz';
import quizData from '../../quizData.json';

describe('Quiz Component', () => {
  const setup = () => {
    const props = {
      quizData
    }

    return shallow(<Quiz {...props} />);
  };

  const wrapper = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call showNextQuestion method', () => {
    const handleSpy = jest.spyOn(wrapper.instance(), 'showNextQuestion');
    wrapper.instance().showNextQuestion();
    expect(handleSpy).toHaveBeenCalled();
  });

  it('should call handleResetClick method', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleResetClick');
    wrapper.instance().handleResetClick();
    expect(spy).toHaveBeenCalled();
  });
});
