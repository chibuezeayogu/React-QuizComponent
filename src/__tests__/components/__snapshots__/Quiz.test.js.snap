// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Component should render correctly 1`] = `
<div
  className="container"
>
  <QuizQuestion
    quizQuestion={
      Object {
        "answer": "7",
        "answerOptions": Array [
          "5",
          "6",
          "7",
          "8",
        ],
        "id": 1,
        "instructionText": "How many continents are there on Planet Earth?",
      }
    }
    showNextQuestionHandler={[Function]}
  />
</div>
`;
